subsys/bluetooth/controller/ll_sw/isoal.c

```diff
@@ -1814,18 +1885,27 @@ static isoal_status_t isoal_tx_framed_produce(struct isoal_source *source,
 		 * 3.1 Time_Offset in framed PDUs :
 		 * The Time_Offset shall be a positive value.
 		 */
-		if (actual_grp_ref_point <= tx_sdu->time_stamp) {
+		if (!isoal_get_time_diff(tx_sdu->time_stamp, actual_grp_ref_point, &time_diff) ||
+			time_diff == 0) {
 			/* Advance target to next event */
 			actual_event++;
-			actual_grp_ref_point += session->iso_interval * ISO_INT_UNIT_US;
+			actual_grp_ref_point = isoal_get_wrapped_time(actual_grp_ref_point,
+							session->iso_interval * ISO_INT_UNIT_US);
 
 			/* Set payload number */
 			pp->payload_number = actual_event * session->burst_number;
 		}
 
 		/* Calculate the time offset */
-		LL_ASSERT(actual_grp_ref_point > tx_sdu->time_stamp);
-		time_offset = actual_grp_ref_point - tx_sdu->time_stamp;
+		time_diff_valid = isoal_get_time_diff(tx_sdu->time_stamp,
+					actual_grp_ref_point, &time_diff);
+		LL_ASSERT(time_diff_valid);
+		LL_ASSERT(time_diff > 0);
+		/* Time difference should be less than the maximum possible
+		 * time-offset of 24-bits.
+		 */
+		LL_ASSERT(time_diff <= 0x00FFFFFF);
```

Comment:
`should` and `assert` doesn't seem to go hand in hand. Should this have been a warning? Or should it have been a `shall`?

Reply:
If the time difference exceeds 24-bits there is something very wrong in either the SDU's timestamp or its target event's reference point. So should / shall ---> must and I would say that the assert is valid in this case.